src/charm.py

```diff
@@ -65,6 +72,14 @@ def __init__(self, *args):
             self._gnb_identity.on.fiveg_gnb_identity_available,
             self._configure_sdcore_nms,
         )
+        self.framework.observe(
+            self.on[GNB_IDENTITY_RELATION_NAME].relation_broken,
+            self._on_fiveg_gnb_identity_relation_broken,
```

Comment:
Couldn't we simply call the `_configure_sdcore_nms` handler instead of creating custom ones? This applies to the other observe as well

Reply:
I thought about it and I think it would be simpler but my concern was about the 2 conditions we have over the `sdcore_management` relation. If it is not properly set, the gnb/upf config files would not be updated.
But we can also say that if we do not have the `sdcore_management` relation, the NMS would not work and the content on the gnb/upf config files is no longer relevant?

Reply:
Should we then update the config files before checking if the nms relation is created in `configure_sdcore_nms` to make sure it's maintained properly?

Reply:
I support Guillaume's opinion here. I think we could use `configure_sdcore_nms` as a main call back.